homeassistant/components/tessie/binary_sensor.py

```diff
@@ -117,13 +117,13 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the Tessie binary sensor platform from a config entry."""
-    coordinators = hass.data[DOMAIN][entry.entry_id]
+    data = hass.data[DOMAIN][entry.entry_id]
```

Comment:
The reason for the data model and renaming the variable is because in the future I want to change the library to one that's classed and can be also passed through the data.

Reply:
The coordinator(s) should hold the data which you now embedded into `TessieVehicle` class so I don't understand what other information you would put in `TessieVehicle`?

Reply:
I meant hass.data not coordinator.data.

In the future I expect there will be two coordinators and an API class in the data model.

Reply:
Shouldn't the api class be in the library?

Reply:
The library currently has no class, which has caused some issues and complexity with the integration that I plan to address in the future.

Essentially I am planning to do this with the data model:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/advantage_air/models.py#L16

Reply:
Ok. I'm not against so if you want to have it like this resolve these and set it for review and I'll make a proper review.